identity/README.md

```diff
@@ -0,0 +1,279 @@
+}
+
+// Specifies one or more capabilities of the CSI-driver.
+message Capability {
+  // Service contains the type of CSI Service that the CSI-driver provides.
+  message Service {
+    // Type describes a CSI Service that CSI-drivers can support.
+    enum Type {
+      // UNKNOWN indicates that the CSI-driver does not neither provide the CSI
+      // ControllerService or CSI NodeService. The CSI-Addons CO plugin will
+      // most likely ignore the node providing this Identify Service.
+      UNKNOWN = 0;
+
+      // CONTROLLER_SERVICE indicates that the CSI-driver provides RPCs for the
+      // CSI ControllerService.
+      // The presence of this capability determines whether the CSI-Addons CO
+      // plugin can invoke RPCs that require access to the storage system,
+      // similar to the CSI Controller (provisioner).
+      CONTROLLER_SERVICE = 1;
+
+      // NODE_SERVICE indicates that the CSI-driver provides RPCs for the CSI
+      // NodeService.
+      // The presence of this capability determines whether the CSI-Addons CO
+      // plugin can invoke RPCs that require a volume to be staged/attached to
+      // the node.
+      NODE_SERVICE = 2;
+    }
+    // type contains the Type of CSI Service that the CSI-driver supports.
+    Type type = 1;
+  }
+
+  // Additional CSI-Addons operations will need to be added here.
+
+  oneof type {
+    // Service or operation that the CSI-driver supports.
+    Service service = 1;
+
+    // Additional CSI-Addons operations need to be appended to this list.
+  }
+}
+```
+
+#### GetCapabilities Errors
+
+If the CSI-driver is unable to complete the GetCapabilities call successfully,
+it MUST return a non-ok gRPC code in the gRPC status.
+
+### `Probe`
+
+The primary utility of the Probe RPC is to verify that the CSI-driver is in a
+healthy and ready state. If an unhealthy state is reported, via a non-success
+response, a CSI-Addons CO plugin MAY take action with the intent to bring the
+CSI-driver to a healthy state. Such actions MAY include, but SHALL NOT be
+limited to, the following:
+
+* Restarting the CSI-driver container, or
+* Notifying the CSI-driver supervisor.
+
+The CSI-driver MAY verify that it has the right configurations, devices,
+dependencies and drivers in order to run and return a success if the validation
+succeeds.
+The CSI-Addons CO plugin MAY invoke this RPC at any time.
+The CSI-Addons CO plugin MAY invoke this call multiple times with the
+understanding that a CSI-driver's implementation MAY NOT be trivial and there
+MAY be overhead incurred by such repeated calls.
+The SP SHALL document guidance and known limitations regarding a particular
+CSI-driver's implementation of this RPC.
+For example, the SP MAY document the maximum frequency at which its Probe
+implementation SHOULD be called.
+
+```protobuf
+// ProbeRequest is sent to the CSI-driver to confirm that it can respond to
+// requests from the CSI-Addons CO plugin.
+message ProbeRequest {
```

Comment:
are these RPC calls idempotent ? I failed to see a mention about that, so asking.

Reply:
yes, there are no procedures in the identity service that write or otherwise affect the CSI-driver configuration or volumes it maintains.

Reply:
sure, I was mainly asking this in general , ie including other node and controller servers.. if we are expecting that all these are idempotent calls, it is good to mention it.